src/main/java/com/gojek/beast/sink/bq/BqSink.java

```diff
@@ -29,48 +29,93 @@
     private final BigQuery bigquery;
     private final TableId tableId;
     private final BQResponseParser responseParser;
-    private final BQErrorHandler errorHandler; // handler instance for BQ related errors
     private final BQRow recordInserter;
+    private final ErrorWriter errorWriter;
 
     private final Stats statsClient = Stats.client();
 
+    private final BQResponseFilter retryableInsertsFilter;
+    private final BQResponseFilter oobInsertsFilter;
+    private final BQResponseFilter unhandledInsertsFilter;
+
     @Override
     public Status push(Records records) {
-        InsertAllResponse response = insertIntoBQ(records);
-        InsertStatus status = new InsertStatus(!response.hasErrors(), response.getInsertErrors());
+        // filter invalid records which cannot be sunk to BQ
+        Map<BQRecordsErrorType, List<Record>> dlqSinkableRecords = new HashMap<>();
+        List<Record> validRecordList = new ArrayList<>();
+        records.forEach(record -> {
+            if (record.getColumns().isEmpty()) {
+                List<Record> errList = dlqSinkableRecords.getOrDefault(BQRecordsErrorType.DESERIALIZE, new ArrayList<>());
+                errList.add(record);
+                dlqSinkableRecords.put(BQRecordsErrorType.DESERIALIZE, errList);
```

Comment:
This is an empty record stored in the underlying datastore. Essentially we are better off to store the actual bytes into the store than not to store anything. In fact we do store the out of bounds messages into the the store. 
Can we follow the same pattern?

Reply:
Sorry didn't get what you mean, when you say store, do you mean GCS or BQ? The current implementation will store bad messages raw bytes to DLQ.

Reply:
Sorry my bad the model has the value stored as raw_bytes. Resolving it now @kushsharma